### ISSUE TITLE
Subscribing to events from the Checks API

### ISSUE THREAD

**Issue by dentarg** [integrations/slack#751]
Doesn't [look like](https://github.com/integrations/slack/blob/04a2cb6de58b705de5c379fff34000c4d26c1270/README.md#repository-activity) it is possible to subscribe to events from the Checks API.

It would be great to be able to do.

This documentation, https://developer.github.com/v3/activity/events/types/, might be relevant.

**Comment by IAmHughes** [comment]
Hey @dentarg , just wanted to let you know that the integration now supports displaying checks in the PR messaging you see in Slack. You can read about the changes we shipped today [on the GitHub Blog](https://github.blog/2019-07-29-whats-new-in-the-github-and-slack-integration/). It's not exactly subscribing just to checks, but it may help unblock your use case.

The key thing to note is that this required some updated permissions on the application, so the owner of any repositories that have the GitHub Slack App installed will need to accept those permissions. If you try to use the command without the permissions, you'll be prompted to update them ❤️.

Thanks for your patience on this! Please re-open this issue if you find that there is an additional use case for subscribing to checks specifically in another way. If there is, if you can give an example of what you'd like to see that would be great!

**Comment by dentarg** [comment]
@IAmHughes I think the use case I had in mind when opening this issue was to get feedback from Travis CI [Cron Jobs](https://docs.travis-ci.com/user/cron-jobs/) in Slack (without having to set up Travis -> Slack integration). However, I'm not sure it is possible to use the Checks API without the context of a pull request. Do you know? :)

**Comment by IAmHughes** [comment]
@dentarg ah okay that's an interesting use case and a bit different than what we shipped. I think this latest update will help some, in that the Checks Suites (see the `checks` tab of a pull request) will report their status when a pull request is unfurled, opened, etc.

- `Checks` tab:
![image](https://user-images.githubusercontent.com/2894107/62218722-9e00b180-b372-11e9-997f-52eef3706ff1.png)


- New functionality:
![image](https://user-images.githubusercontent.com/2894107/62218636-7873a800-b372-11e9-94b2-7d9d8d6021f9.png)

I'll re-open this issue as what you're referring to would be a little different I believe. Thanks for the feedback!

**Comment by scepticulous** [comment]
Hi @dentarg 👋 

>  However, I'm not sure it is possible to use the Checks API without the context of a pull request. Do you know? :)

Checks (CheckRun and CheckSuite) only [require a the head-sha (commit hash)](https://developer.github.com/v3/checks/runs/#parameters), so they are not bound to a pull-request.

So if you want to create a Check based on a commit hash, that should not be a problem.
This in conjunction with the new support for Check related events, you should be able to create a Check without a PR from Travis and see the result in Slack. 
Might be worth giving it a shot.

**Comment by dentarg** [comment]
Hi @scepticulous 

I just checked my repo with [`check_runs_for_ref`](https://github.com/octokit/octokit.rb/blob/v4.14.0/lib/octokit/client/checks.rb#L49-L70) in Octokit and I can see that Travis CI created an check run for the cron job that ran today.

I also used `/github subscribe list features` and checked what my repo is subscribed to: `issues, pulls, deployments, statuses, public, commits:all, releases, comments, branches, reviews`

I take it that `statuses` should include check runs:

https://github.com/integrations/slack/blob/a4e5847b9bce5d4108955144ef2f700ca9036757/lib/models/subscription.js#L8

We have also approved the "Updated Permissions Request" for the GitHub App Slack (almost two weeks ago).

But I haven't seen any message in Slack that could be the check run. Is there any other criteria that needs to be met?   Other events like comments, new pull requests, deployments is posted correctly to Slack. Do I need to re-subscribe?

**Comment by scepticulous** [comment]
Hi @dentarg, 
just to make sure I have all the information.
Did you check out this [part of the PR+Status announcement](https://github.blog/2019-07-29-whats-new-in-the-github-and-slack-integration/#view-checks-on-pull-requests)?

Checks and Statuses are part of the PR message that is posted to the Slack channel. So in order to reproduce the behavior could you please try the following:

* subscribe to a repo including the `pulls` and `statuses` features (should be the default
* create a PR in the repo
* verify that the PR is posted to that slack channel
* create a Check-Run within 30 (ideally a few) minutes and check if there is a status section at the bottom of the PR message

It should look like the image in the linked blog post. Let me know if this works for you and if not, what exact step failed.

**Comment by dentarg** [comment]
Yes, I did read the blog post, and yes, it works for pull requests. I just did this test:

<img width="398" alt="Screenshot 2019-08-12 at 16 22 52" src="https://user-images.githubusercontent.com/42626/62872312-7e2b9f00-bd1d-11e9-8032-abb877798833.png">

> * create a Check-Run within 30 (ideally a few) minutes and check if there is a status section at the bottom of the PR message

If I do this, will it update the same message, or post a new message?

**Comment by scepticulous** [comment]
It will update the existing message for a duration of max 30 minutes AFAIR. 
(I would have to check for more details)

**Comment by dentarg** [comment]
@scepticulous Okay, but how is this related to check runs without pull requests? Are you just making sure this part is working?

**Comment by scepticulous** [comment]
Hey @dentarg,

sorry for the confusion. When I initially answered your question about Checks without PRs, I was only talking about CheckRuns independent of the Slack context.
After double-checking the implementation, the Slack integration only handled Checks that are attached to a PR.

https://github.com/integrations/slack/blob/33ed9a24dd415f40b164bcc9c1d4eee563d4626f/lib/activity/pull-requests.js#L99-L116

I assume that is not what you needed and I might have added to your confusion. Sorry about that.
Am I correct, that you are looking for a way to receive check-results without a PR in Slack?
If so, we have to consider this a feature request, since the last ship only added PR-related status/checks capabilities.

**Comment by dentarg** [comment]
> Am I correct, that you are looking for a way to receive check-results without a PR in Slack?

Yes!

> If so, we have to consider this a feature request, since the last ship only added PR-related status/checks capabilities.

Sure, I guess that's why @IAmHughes re-opened this issue.

**Comment by mastoj** [comment]
Why does it update the existing message? I would rather see it creating a new message for every new check run since the opening of the PR can be far back in the slack history and harder to find. Also, my initial PR passed all the checks, but when I now committed a bug that doesn't update the message.

It might be that it is a good reason for not creating a new message, but I do think that would make more sense. Preferably it should also contain a link to the details for the check run.

**Comment by case** [comment]
> I also agree that it should be disconnected from PR, especially now with actions that you have running for a branch that might not have a PR associated with it yet.

Or for Actions that run on a cron schedule, for example.

**Comment by fortum-vaanavil** [comment]
We are running checks (via Actions) in PRs but also on `master`. Our build pipeline only triggers after checks in `master` pass and the current behavior of the Slack integration causes confusion as to why didn't the build pipeline start. You can see e.g. "1 new commit pushed to master" and would expect the next message to come from the build pipeline, but if the checks fail, nothing's reported to Slack and the build pipeline never triggers.

**Comment by caseywebdev** [comment]
+1 for the same issue @fortum-vaanavil described. The checks support on PRs is great, but we run the same checks after commits are merged (or pushed directly) to master and those are not currently possible to display with this integration AFAIK because they're outside a PR. For now we're manually posting to Slack at the start/end of a GH action only when the branch is master as a workaround.
